src/tactic/ui/tools/repo_browser_wdg.py

```diff
@@ -1121,8 +1118,103 @@ def add_top_behaviors(my, top):
             }
 
         }
+         
+        spt.repo_browser.delete_empty_folder = function(relative_dir) {
+            if (!relative_dir) return;
+
+            var class_name = 'tactic.ui.tools.RepoBrowserActionCmd';
+            var kwargs = {
+                action: 'delete_folder',
+                relative_dir: relative_dir
+            }
+            try {
+                server.execute_cmd(class_name, kwargs);
+            }
+            catch(err) {
+                var error = spt.exception.handler(err);
+                var message = "Selected snapshots deleted but an error occured when removing the directory ["+relative_dir+"]";
+                message = message + "Error: " + error;
+                spt.alert(message);
+            }
+        }
+
+        spt.repo_browser.delete_assets = function(bvr) {
+            var server = TacticServerStub.get();
+            var activator = spt.smenu.get_activator(bvr);
+            
+            if ([true, "True", "true"].indexOf(bvr.single_asset_mode) > -1) {
```

Comment:
did u do a search thru all the js to ensure it needs to check for parent_mode == 'single_asset' or 'single_file' or 'single_search_type' ?.. Can all 3 modes be explained in hte ARGS_KEYS?